### PR TITLE
Cassandra 17821 - Add ability to log load profiles at fixed intervals

### PR DESCRIPTION
**Pull request by jmckenzie-dev**

### CODE REVIEW

File: NEWS.txt [hunk 1]
```diff
@@ -74,10 +74,12 @@ New features
       - Maximum replication factor
       - Whether DROP KEYSPACE commands are allowed.
     - It is possible to list ephemeral snapshots by nodetool listsnaphots command when flag "-e" is specified.
+    - Added a new flag to `nodetool profileload` and JMX endpoint to set up recurring profile load generation on specified
+      intervals (see CASSANDRA-17821)
 
 Upgrading
 ---------
-    - Emphemeral marker files for snapshots done by repairs are not created anymore, 
+    - Ephemeral marker files for snapshots done by repairs are not created anymore,
```
Comment (smiklosovic) [inline NEWS.txt:82]: thanks!

File: src/java/org/apache/cassandra/metrics/SamplingManager.java [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+            activeSamplingTasks.put(jobId, fut);
+        };
+    }
+
+    /**
+     * Finish the sampling and begin a new one immediately after.
+     *
+     * NOTE: Do not call this outside the context of {@link this#createSamplingBeginRunnable}, as we need to preserve
+     * ordering between a "start" and "end" runnable
+     * @return
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/metrics/SamplingManager.java:230]: nit: this line seems not necessary or incomplete.

File: src/java/org/apache/cassandra/metrics/SamplingManager.java [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+        Set<JobId> allJobIds = jobIds();
+        // There is a schedule that works on all tables. Overlapping guaranteed.
+        if (allJobIds.contains(JobId.ALL_KS_AND_TABLES) || (!allJobIds.isEmpty() && jobId.equals(JobId.ALL_KS_AND_TABLES)))
+            return false;
+        // there is an exactly duplicated schedule
+        else if (allJobIds.contains(jobId))
+            return false;
+        else
+            // make sure has no overlapping tables under the keyspace
+            return !allJobIds.contains(new JobId(jobId.keyspace, null));
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/metrics/SamplingManager.java:175]: There is `JobId.createForAllTable(keyspace)` can replace `new JobId` here

**Review by yifan-c — APPROVED** [review]
a few nits

File: src/java/org/apache/cassandra/metrics/SamplingManager.java (hunk 2 above)
Comment (jmckenzie-dev) [reply src/java/org/apache/cassandra/metrics/SamplingManager.java:230]: Yep! Removed.

File: src/java/org/apache/cassandra/metrics/SamplingManager.java (hunk 3 above)
Comment (jmckenzie-dev) [reply src/java/org/apache/cassandra/metrics/SamplingManager.java:175]: 👍

**Pull request closed without merging**